### ISSUE TITLE
Color diffusion mode gives less useful results in flamechart mode

### ISSUE THREAD

**Issue by itamarst** [jonhoo/inferno#272]
Consider the following profiling result:

```
bottleneck 20
interlude 20
bottleneck 20
interlude 20
bottleneck 20
```

In flamegraph mode with `--colordiffusion`, the bottleneck is darker red, since it is 60% of time spent vs 40%:

![Screenshot from 2022-10-30 08-57-01](https://user-images.githubusercontent.com/3266662/198879699-a4ad56ed-52ab-4383-88ef-17305395f418.png)

In flamechart mode, however, the segments all have the same red saturation:

![Screenshot from 2022-10-30 08-57-01](https://user-images.githubusercontent.com/3266662/198879716-cc5edbaf-a6f2-4ac0-b52c-e96f8829c54a.png)

Insofar as the goal of the color mode is to highlight more significant parts of the code, this is wrong: the relative saturation should be the same as in the flamegraph. Just because the bottleneck is chopped up doesn't mean it has ceased to be the bottleneck, it should still be redder.

**Comment by jonhoo** [comment]
I agree, although I also suspect this'll be quite tricky to fix in chart mode. You'd need a sort of two-pass thing where we first scan for how expensive each function is in aggregate (by basically doing a non-chart plot and now drawing it), and then carrying that information forward into chart mode.

**Comment by itamarst** [comment]
Yeah, or a trie-like thing.
